erdpy/delegation/staking_provider.py

```diff
@@ -0,0 +1,141 @@
+import binascii
+from typing import Any
+from os import path
+
+from erdpy.accounts import Account
+from erdpy.config import MetaChainSystemSCsCost
+from erdpy.utils import read_json_file_validators, parse_keys
+from erdpy.validators.core import estimate_system_sc_call
+from erdpy.wallet.pem import parse_validator_pem
+from erdpy.wallet.signing import sign_message_with_bls_key
+
+DELEGATION_MANAGER_SC_ADDRESS = "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqylllslmq6y6"
+
+
+def prepare_args_for_create_new_staking_contract(args: Any):
+    args.data = 'createNewDelegationContract'
+    args.data += '@' + string_to_hex_bytes(str(args.total_delegation_cap))
+    args.data += '@' + string_to_hex_bytes(str(args.service_fee))
+
+    args.receiver = DELEGATION_MANAGER_SC_ADDRESS
+
+    if args.estimate_gas:
+        args.gas_limit = estimate_system_sc_call(args, MetaChainSystemSCsCost.DELEGATION_MGR_OPS, 3)
+
+
+def string_to_hex_bytes(number_str: str) -> str:
+    num_of_bytes = 1
+    if len(number_str) > 2:
+        num_of_bytes = int(len(number_str) / 2)
+    int_str = int(number_str)
+    int_bytes = int_str.to_bytes(num_of_bytes, byteorder="big")
+    bytes_str = binascii.hexlify(int_bytes).decode()
+    return bytes_str
+
+
+def prepare_args_for_add_nodes(args: Any):
```

Comment:
Another example of "bad programming" is that we are continuously passing the same arguments all over the place instead of grouping the functions in a specialized class